# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [3]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [5]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4         V5        V6        V7       
1 0.4676077 0.2146855 0.3834354 0.95043219 0.5354503 0.5620537 0.5769928
2 0.5038885 0.3889348 0.8842110 0.09738971 0.1267551 0.2547298 0.4543927
  V8        V9        V10        V11       V12       V13       V14      
1 0.2471133 0.4626011 0.13514525 0.3366562 0.7788851 0.1898025 0.2228375
2 0.8532304 0.4161341 0.01044126 0.1169245 0.1398768 0.7942234 0.4055758
  V15       V16       V17       V18       V19       V20      
1 0.5763071 0.8983577 0.1134810 0.3368753 0.0366940 0.2703048
2 0.5269538 0.4044039 0.7214336 0.9192030 0.5001592 0.4028239

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>%
gather(Key, Value, V1:V20) 

head(df2)

Key Value    
1 V1  0.4676077
2 V1  0.5038885
3 V2  0.2146855
4 V2  0.3889348
5 V3  0.3834354
6 V3  0.8842110

## Assignment 2

We will generate some data:

In [7]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [9]:
df5 <- full_join(df3, df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook